File: DemoMethods/Basic/StaticQuery.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using DemoMethods.Indexes;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
        [HttpGet]
        [Demo("Static Query", DemoOutputType.Flatten, demoOrder: 70)]
        public object StaticQuery(string highPrice = "500", string delayDays = "35")
        {
            var HighPrice = int.Parse(highPrice);
            var DelayDays = int.Parse(delayDays);

            //GetImportantOrdersWithIssues - High Price Orders and Delayed Orders
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                var problemtaticOrders = session
                    .Query<CostlyOrders.Result, CostlyOrders>()
                    .Where(x => x.Price > HighPrice && x.Delay > TimeSpan.FromDays(DelayDays))
                    .OfType<Order>()
                    .ToList();

                return (problemtaticOrders);
            }
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/basic/StaticQuery.java
package net.ravendb.demo.basic;

import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Order;
import net.ravendb.demo.indexes.CostlyOrders;
import net.ravendb.demo.indexes.QCostlyOrders_Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.time.Duration;
import java.util.List;

@Controller
public class StaticQuery {
    @RequestMapping("/Basic/StaticQuery")
    public List<Order> staticQuery(
            @RequestParam(value = "highPrice", defaultValue = "500") int highPrice,
            @RequestParam(value = "delayDays", defaultValue = "35") int delayDays) {

        //GetImportantOrdersWithIssues - High Price Orders and Delayed Orders
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            QCostlyOrders_Result c = QCostlyOrders_Result.result;

            long delayInNanos = Duration.ofDays(delayDays).toNanos() / 100;

            List<Order> problematicOrder = session.query(CostlyOrders.Result.class, CostlyOrders.class)
                    .where(c.price.gt(highPrice).and(c.delay.gt(delayInNanos)))
                    .as(Order.class)
                    .toList();
            return problematicOrder;
        }
    }
}
